### PR TITLE
Troubleshooting

### PR DESCRIPTION
**Pull request by MartinJHammer**
Hi there,

This is my first PR to a public project, so beware ;)

Just wanted to add a troubleshooting section to the docs for known problems with known solutions. I added the first section based on [this issue](https://github.com/redux-offline/redux-offline/pull/229).

If you want the offline trigger example somewhere else (e.g. drop the trouble shooting section), just write where else I should put it. Any other corrections are also welcome :)

### CODE REVIEW

File: docs/troubleshooting/offline-queue-trigger.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
+# Offline queue doesn't trigger on network status change
```
Comment (calumpeak) [inline docs/troubleshooting/offline-queue-trigger.md:1]: General housekeeping for this PR to align it with the rest of the docs/codebase.

- Can we remove all the typescript decoration and prefer javascript
- Remove angular decorations, prefer basic redux setup please
- Remove any middleware that isn't redux/redux-offline specific (e.g. redux-observable)
- Please make note that this is if you're using `createOffline` and want to compose yourself not the default setup.

I think that only a working example and highlighting the change as you have done is needed. Should aim to keep it short and sweet.

**Review by calumpeak — CHANGES_REQUESTED** [review]

File: docs/troubleshooting/offline-queue-trigger.md (hunk 1 above)
Comment (sorodrigo) [reply docs/troubleshooting/offline-queue-trigger.md:1]: I feel like we can include the redux-observable setup in a separate chapter called `Redux-Observable` inside the recipes section. And BTW thanks for contributing! 🎉

**Comment by MartinJHammer** [comment]
Hi there,

I almost forgot all about this. Life has been busy and I haven't found the time to finish. Even now I don't have the time. Sorry about that! 

I'll close this pull request for now. If I get time in the future I'll open a new one. Hope that is the correct way to do it :)

**Comment by sorodrigo** [comment]
Hello, it’s a shame that you deleted the branch. I think that your PR was useful, I’m just too busy right now and can’t add the final changes. I will try add what you wrote down eventually.

**Comment by MartinJHammer** [comment]
Ack! I was way too quick on the trigger there. I feel kinda bad about that :( Still not too sure on proper "open source" etiquette.

To make up for it, I'll rewrite the guide as good as I can tomorrow (hopefully) and post it here, so it will be easy for you to correct and add in the proper place.

Hope that will help :)

**Comment by MartinJHammer** [comment]
Okay, I finally took myself by the ear and recreated the guide. I will post it below. I still feel really bad about aborting the whole thing, so if you guys want, I can reopen this pull request or open a new one and do the merge. Just tell me the corrections to below + where you want it to go, and I'll do it this time :)
<pre>
# How to fix offline queue not executing when coming back online
If you have actions in your redux-offline queue, all of them should be dispatched as soon as your app goes online again. If you experience the queue *doesn't* execute, then this guide will explain how to fix it, and why it works.

## TL;DR - The fix
For you in a hurry; The error is caused by your redux store getting configured incorrectly.

When composing your store's enhances, the offlineEnhancer ***MUST*** come before you apply middleware.

**WRONG**
```javascript
// redux-configuration.js

const reduxOfflineConfiguration = { /*config stuff*/ };
const initialState = {};
const offlineEnhancer = createOffline(reduxOfflineConfiguration);
function rootReducer() {
  // Root reducer code
}

const store = createStore(
    offlineEnhancer.enhanceReducer(rootReducer),
    initialState,
    compose(
        applyMiddleware(offlineEnhancer.middleware),
        offlineEnhancer.enhanceStore //<-- The offline enhancer comes in last. Your queue will not execute when getting back online :(
    )
);
```

**CORRECT**
```javascript
// redux-configuration.js

const reduxOfflineConfiguration = { /*config stuff*/ };
const initialState = {};
const offlineEnhancer = createOffline(reduxOfflineConfiguration);
function rootReducer() {
  // Root reducer code
}

const store = createStore(
    offlineEnhancer.enhanceReducer(rootReducer),
    initialState,
    compose(
        offlineEnhancer.enhanceStore, // <-- The offline enhancer comes first. Your queue will execute just fine :)
        applyMiddleware(offlineEnhancer.middleware)
    )
);
```

## Why
The reason why the above store config needs to be as described above is very well explained by github user j8seangel in this redux-offline issue: "[Fix offline not starting after rehydrate](https://github.com/forest-watcher/forest-watcher/pull/303)" (He's also the original author of above fix). I'll quote his answer here:

>Order matters, swapping two words could fix a critical bug 🎉
>
>The offline middleware wasn't aware of the outbox actions after rehydrate until another action was dispatched, so when the app starts and it doesn't dispatch any other action the offline wouldn't start and the actions would remain there forever.
>
>Now the middleware is applied first and all of the actions go through it, including the rehydrate one.
>
>Reference: https://redux.js.org/api-reference/compose
>Key: `Composes functions from right to left.`

If this doesn't fix your queue not executing, open an issue :)

Happy programming!
</pre>

**Comment by sorodrigo** [comment]
Hey!! Thank you so much!! Don’t feel bad, on the opposite I really appreciate this. I don’t want to take more time from you, and at the same time your contribution should be reflected on the history of this repo. So don’t feel obligated but at the same time if you want to open the PR feel free to do so. Otherwise I can add this later and give you some credit for it 🙂.

**Comment by MartinJHammer** [comment]
Thanks for the pep :) I'll open a PR this saturday. I have reread the guide, and I can see I need to edit it a bit + add the note echoes221   mentioned (Please make note that this is if you're using `createOffline` and want to compose yourself not the default setup.)

**Comment by MartinJHammer** [comment]
I reopened this pull request [here](https://github.com/redux-offline/redux-offline/pull/300) :)
